### PR TITLE
MacOS `make generate` fix - regenerate certain mocks using source mode instead of reflect mode

### PR DESCRIPTION
**Pull request by kimorris27**
### Which issue this PR addresses:

No Jira

### What this PR does / why we need it:

I found after updating my OS to Sonoma 14.7, which included updating the Command Line Tools for Xcode from 15.3 to 16.0, that `make generate` was failing for certain packages and was also outputting some linker warnings. I can't explain exactly why it works, but I found that having `mockgen` use source mode instead of reflect mode for the failing packages fixed things. The linker warnings about a deprecated `-ld_classic`, which were noted in the [CLT for Xcode 16.0 release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-16-release-notes#Linking), also went away, which is interesting to note.

I could try to dig into the reasoning for this change further, but I don't think it'd be a great use of time.

I've encountered multiple issues related to reflect mode in my work recently, which makes me wonder if we should switch to using source mode throughout the repository.

### Test plan for issue:

Tested locally to confirm that it fixes `make generate` on my Mac and unit test still work, and then ran `make ci-rp` to make sure it didn't break anything in the containerized process either.

### Is there any documentation that needs to be updated for this PR?
 
N/A

### How do you know this will function as expected in production? 

Not a production change; only affects generation of mock interfaces that are used in unit tests.

### CODE REVIEW

**Comment by kimorris27** [comment]
Skipping E2E since this PR only impacts unit tests.

**Review by tsatam — APPROVED** [review]
LGTM, I tested `make generate` on Linux and it works fine and doesn't create a diff, our CI checks back that up too. 

I'd be incredibly disappointed if somehow changing our mock generation impacts production flows in any way, but out of an abundance of caution I say we should wait for green E2E on this before merge.

**Pull request merged**
